src/App.js

```diff
@@ -1,23 +1,51 @@
-import logo from './logo.svg';
-import './App.css';
+import axios from 'axios';
+import React, {useCallback, useEffect, useState} from "react";
 
 function App() {
+
+  const [data, setData] = useState({})
+  const [city, setCity] = useState("")
+  //const [ex, setEx] = useState({})
+  const [url, setUrl] = useState("")
+
+  const fetchCity = useCallback(() =>{
```

Comment:
Part of the logic, you should export all function which are based on the logic of app in another folder. It will improve the global scalibility of the app